pyiron_base/storage/hdfio.py

```diff
@@ -120,23 +136,16 @@ def _to_object(hdf, class_name=None, **kwargs):
         raise ValueError(
             "Object type in hdf5-file must be identical to input parameter"
         )
-    class_name = class_name or hdf.get("TYPE")
-    class_path = class_name.split("<class '")[-1].split("'>")[0]
-    class_convert_dict = {  # Fix backwards compatibility
-        "pyiron_base.generic.datacontainer.DataContainer": "pyiron_base.storage.datacontainer.DataContainer",
-        "pyiron_base.generic.inputlist.InputList": "pyiron_base.storage.inputlist.InputList",
-        "pyiron_base.generic.flattenedstorage.FlattenedStorage": "pyiron_base.storage.flattenedstorage.FlattenedStorage",
-    }
-    if class_path in class_convert_dict.keys():
-        class_name_new = "<class '" + class_convert_dict[class_path] + "'>"
-        class_object = _import_class(class_name_new)
-    elif not class_path.startswith("abc."):
-        class_object = _import_class(class_name)
+    type_field = class_name or hdf.get("TYPE")
+    module_path, class_name = _extract_module_class_name(type_field)
+
+    if not module_path.startswith("abc."):
```

Comment:
We talked about this briefly as I had concerns that this was about dynamically defined classes but the `abc` there can be somehow overloaded. However, I don't remember why it's _actually_ here. It's also not clear to me from reading what the intent is. Can you add a `#comment` about why we want to do this, i.e. what case does it cover?

Reply:
Will add a comment to the following effect.  From my understanding, this is because [dynamic jobs](https://github.com/pyiron/pyiron_base/blob/e6cca773c11bfade35b445b76f83d8225552f0c7/pyiron_base/jobs/dynamic.py#L132) (which were patched in quickly about this time last year, if I remember correctly).  One can add scripts into a magic location in the resources and then the job factory creates job classes dynamically (one class for each script).  Apparently when writing them to HDF, nothing is handled and so TYPE ends up being the `abc.` path, that means when reading them we have to guess that all objects starting with `abc.` are dynamic jobs and look this up in the `JOB_DYN_DICT`.

Afaik, this is nowhere documented in more detail, or did I miss this somewhere @jan-janssen ?